oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderJwkSupport.java

```diff
@@ -62,27 +53,23 @@
  * <p>
  * <b>NOTE:</b> This implementation uses the Nimbus JOSE + JWT SDK internally.
  *
+ * @deprecated Use {@link NimbusJwtDecoder} or {@link JwtDecoders} instead
+ *
  * @author Joe Grandja
  * @author Josh Cummings
  * @since 5.0
  * @see JwtDecoder
+ * @see NimbusJwtDecoder
  * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7519">JSON Web Token (JWT)</a>
  * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7515">JSON Web Signature (JWS)</a>
  * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7517">JSON Web Key (JWK)</a>
  * @see <a target="_blank" href="https://connect2id.com/products/nimbus-jose-jwt">Nimbus JOSE + JWT SDK</a>
  */
 public final class NimbusJwtDecoderJwkSupport implements JwtDecoder {
```

Comment:
I think if we have done things well in `NimbusJwtDecoder` then we should be able to delete more of this class. For example, I don't think we should need `RestOperationsResourceRetriever`. A way to minimize this would be to have a member of the `Builder` and a member of the `NimbusJwtDecoder` If a setter is invoked, it updates the builder and then builds a new delegate `NimbusJwtDecoder`